internal/metrics/versions.go

```diff
@@ -27,25 +27,26 @@ func UpdateDefaultVersions(version, commitid string) {
 // ResolveVersions is a best-effort to resolve current pod's versions info
 func ResolveVersions(clnt *kclient) (Versions, error) {
 	var err error
+	var status error
 	vers := Versions{
 		Version:  defaultVersions.Version,
 		CommitID: defaultVersions.CommitID,
 	}
 	if clnt != nil {
 		vers.SambaImage, err = resolveSambaImage(clnt)
 		if err != nil {
-			return vers, err
+			status = err
 		}
 	}
 	vers.SambaVersion, err = resolveSambaVersion()
-	if err != nil {
-		return vers, err
+	if err != nil && status != nil {
+		status = err
 	}
 	vers.CtdbVersion, err = resolveCtdbVersion()
-	if err != nil {
-		return vers, err
+	if err != nil && status != nil {
+		status = err
 	}
-	return vers, nil
+	return vers, status
```

Comment:
why not use a `defer`statement for the return at the top of the function?

Reply:
Hmm, it looks to me like it would make an unnecessary complication to the code, and I don't think  it is common by Go developers to use it for such cases.

Reply:
I am unclear on how defer would help here. I do find `status` a bit odd. Maybe consider a pattern like:
```golang
var err, innerErr error
vers.foo, innerErr = doThing()
err = errors.Join(err, innerErr)
vers.bar, innerErr = doAnotherThing()
err = errors.Join(err, innerErr)
/* ... and so on ... */
return vers, err
```

I think that would simplify the function, but I won't demand it.

Reply:
Agreed. `errors.Join` would make the code cleaner. Will fix.